From 263b65d0fc8f92375cd5a8f0c258b8bbc47513ee Mon Sep 17 00:00:00 2001
From: Jason Montleon <jmontleo@redhat.com>
Date: Tue, 28 Oct 2014 09:13:57 -0400
Subject: [PATCH] etnaviv patch

---
 drivers/staging/etnaviv/etnaviv_buffer.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/staging/etnaviv/etnaviv_buffer.c b/drivers/staging/etnaviv/etnaviv_buffer.c
index 0ce1e4b..b6d28e3 100644
--- a/drivers/staging/etnaviv/etnaviv_buffer.c
+++ b/drivers/staging/etnaviv/etnaviv_buffer.c
@@ -135,8 +135,8 @@ static void etnaviv_buffer_dump(struct etnaviv_gpu *gpu,
 	u32 size = obj->base.size;
 	u32 *ptr = obj->vaddr + off;
 
-	dev_info(gpu->dev, "virt %p phys 0x%08x free 0x%08x\n",
-			ptr, obj->paddr + off, size - len * 4 - off);
+    dev_info(gpu->dev, "virt %p phys 0x%llx free 0x%08x\n",
+    ptr, (u64)obj->paddr + off, size - len * 4 - off);
 
 	print_hex_dump(KERN_INFO, "cmd ", DUMP_PREFIX_OFFSET, 16, 4,
 			ptr, len * 4, 0);
@@ -203,9 +203,9 @@ void etnaviv_buffer_queue(struct etnaviv_gpu *gpu, unsigned int event,
 		cmd->offset = submit->cmd[i].offset + submit->cmd[i].size;
 
 		if (drm_debug & DRM_UT_DRIVER)
-			pr_info("stream link from buffer %u to 0x%08x @ 0x%08x %p\n",
+			pr_info("stream link from buffer %u to 0x%08x @ 0x%llx %p\n",
 				i, link_target,
-				cmd->paddr + cmd->offset * 4,
+				(u64)cmd->paddr + cmd->offset * 4,
 				cmd->vaddr + cmd->offset * 4);
 
 		/* jump back from last cmd to main buffer */
@@ -229,7 +229,7 @@ void etnaviv_buffer_queue(struct etnaviv_gpu *gpu, unsigned int event,
 		pr_info("link op: %p\n", lw);
 		pr_info("link addr: %p\n", lw + 1);
 		pr_info("addr: 0x%08x\n", link_target);
-		pr_info("back: 0x%08x\n", buffer->paddr + (back * 4));
+		pr_info("back: 0x%llx\n", (u64)buffer->paddr + (back * 4));
 		pr_info("event: %d\n", event);
 	}
 
