diff -ur xserver-xorg-video-imx-viv-3.10.17-1.0.0/DRI_1.10.4/src/drm.h xserver-xorg-video-imx-viv-3.10.17-1.0.0_libdrm/DRI_1.10.4/src/drm.h
--- xserver-xorg-video-imx-viv-3.10.17-1.0.0/DRI_1.10.4/src/drm.h	2014-04-02 14:37:09.000000000 +0200
+++ xserver-xorg-video-imx-viv-3.10.17-1.0.0_libdrm/DRI_1.10.4/src/drm.h	2014-05-09 22:46:45.005756929 +0200
@@ -617,7 +617,7 @@
     __u64 value;
 };
 
-#include <drm/drm_mode.h>
+#include <libdrm/drm_mode.h>
 
 #define DRM_IOCTL_BASE            'd'
 #define DRM_IO(nr)            _IO(DRM_IOCTL_BASE,nr)
diff -ur xserver-xorg-video-imx-viv-3.10.17-1.0.0/EXA/src/makefile.linux xserver-xorg-video-imx-viv-3.10.17-1.0.0_libdrm/EXA/src/makefile.linux
--- xserver-xorg-video-imx-viv-3.10.17-1.0.0/EXA/src/makefile.linux	2014-04-02 14:37:09.000000000 +0200
+++ xserver-xorg-video-imx-viv-3.10.17-1.0.0_libdrm/EXA/src/makefile.linux	2014-05-09 22:46:37.235304179 +0200
@@ -54,7 +54,7 @@
 LOCAL_CFLAGS += -fvisibility=hidden -fPIC -DPIC -DFREESCALE_EXTENSION
 LOCAL_CFLAGS += -I. -I.. -I$(sysroot)/usr/include -I$(sysroot)/usr/include/xorg -I$(sysroot)/usr/include/$(PIXMAN)
 LOCAL_CFLAGS += -I./vivante_gal -I./vivante_fbdev -I./vivante_util -I./vivante_exa -I./vivante_extension
-LOCAL_CFLAGS += -I$(sysroot)/usr/include/drm
+LOCAL_CFLAGS += -I$(sysroot)/usr/include/libdrm
 
 LOCAL_LFLAGS += -fPIC
 LOCAL_LFLAGS += -L$(sysroot)/usr/lib -lm -ldl -lGAL -Wl,-soname -Wl,vivante_drv.so
diff -ur xserver-xorg-video-imx-viv-3.10.17-1.0.0/FslExt/src/makefile.linux xserver-xorg-video-imx-viv-3.10.17-1.0.0_libdrm/FslExt/src/makefile.linux
--- xserver-xorg-video-imx-viv-3.10.17-1.0.0/FslExt/src/makefile.linux	2014-04-02 14:37:09.000000000 +0200
+++ xserver-xorg-video-imx-viv-3.10.17-1.0.0_libdrm/FslExt/src/makefile.linux	2014-05-09 22:47:01.606717829 +0200
@@ -27,7 +27,7 @@
 # macro FREESCALE_EXTENSION is reserved for future use
 LOCAL_CFLAGS += -fPIC 
 LOCAL_CFLAGS += -I. -I.. -I$(sysroot)/usr/include -I$(sysroot)/usr/include/xorg
-LOCAL_CFLAGS += -I$(sysroot)/usr/include/drm
+LOCAL_CFLAGS += -I$(sysroot)/usr/include/libdrm
 
 LOCAL_LFLAGS += -fPIC
 LOCAL_LFLAGS += -L$(sysroot)/usr/lib -lGAL -lXext -lX11 -lc -Wl,-soname -Wl,libfsl_x11_ext.so
